### DISCUSSION TITLE
Include classvar in exported model

### DISCUSSION THREAD

**Discussion by mikkelam** [pydantic/pydantic#2505] [Question] [answered]
Hi all,

I'm trying to model an object like so:

```
from pydantic import BaseModel
class MyModel(BaseModel):
    status: ClassVar[str] = 'Ready'

```
Where the status is _excluded_ in the schema, but it is _included_ in the exported model. I'm not really sure if what i'm doing is stupid, or should not be done, but would love some feedback. 

Basically, it matches my internal model that should be written to my db, but users are not allowed to input the status at this stage, and thus should always be `'Ready'` initially.

Thanks

**Comment by PrettyWood** [comment, marked as answer]
Hi @mikkelam 
You could do something like
```py
class MyModel(BaseModel):
    status: str = 'Ready'
    qwe: str

    class Config:
        schema_extra = lambda schema: schema['properties'].pop('status')
```
But IMO it's ok to create 2 models. One for your output and one for your internal state that inherits from your output model to just extend the fields
```py
class Output(BaseModel):
    qwe: str

class Internal(Output):
    status: str
```
Hope it helps

**Reply by mikkelam** [reply]
Thanks @PrettyWood. I think the internal model is more clean here, I think I'll go for that
